diff -ruN -x Makefile -x '*.o' -x 'atari800*' -x autom4te.cache -x 'config*' -x atari.ro ../../emusrc/atari800/src/Makefile.in ../atari800/src/Makefile.in
--- ../../emusrc/atari800/src/Makefile.in	2012-09-06 17:28:13.005504300 +0200
+++ ../atari800/src/Makefile.in	2013-03-04 15:17:36.601659000 +0100
@@ -59,6 +59,7 @@
 	sio.o \
 	sysrom.o \
 	util.o \
+	avic.o \
 	@OBJS@
 
 
@@ -84,7 +85,7 @@
 	$(RC) --define WIN32 --define __MINGW32__ --include-dir . $< $@
 
 %.o: %.cpp
-	$(CC) -c -o $@ $(DEFS) -I. $(CFLAGS) $<
+	$(CC) -c -o $@ $(DEFS) -I. $(CFLAGS) -fpermissive $<
 
 %.o: %.S
 	$(CC) -c -o $@ $(DEFS) -I. $(CFLAGS) $<
diff -ruN -x Makefile -x '*.o' -x 'atari800*' -x autom4te.cache -x 'config*' -x atari.ro ../../emusrc/atari800/src/atari.c ../atari800/src/atari.c
--- ../../emusrc/atari800/src/atari.c	2012-11-05 15:06:28.933112200 +0100
+++ ../atari800/src/atari.c	2013-03-04 14:24:15.923590700 +0100
@@ -124,6 +124,7 @@
 #include "Helpers.h"
 #endif /* _WX_ */
 #endif /* __PLUS */
+#include "avic.h"
 #ifdef PBI_BB
 #include "pbi_bb.h"
 #endif
@@ -260,6 +261,7 @@
 #endif
 	/* reset cartridge to power-up state */
 	CARTRIDGE_ColdStart();
+	AVIC_Coldstart();
 	/* set Atari OS Coldstart flag */
 	MEMORY_dPutByte(0x244, 1);
 	/* handle Option key (disable BASIC in XL/XE)
@@ -745,6 +747,7 @@
 		|| !CARTRIDGE_Initialise(argc, argv)
 		|| !CASSETTE_Initialise(argc, argv)
 		|| !PBI_Initialise(argc,argv)
+		|| !AVIC_Initialise(argc, argv)
 #ifdef VOICEBOX
 		|| !VOICEBOX_Initialise(argc, argv)
 #endif
diff -ruN -x Makefile -x '*.o' -x 'atari800*' -x autom4te.cache -x 'config*' -x atari.ro ../../emusrc/atari800/src/avic.c ../atari800/src/avic.c
--- ../../emusrc/atari800/src/avic.c	1970-01-01 01:00:00.000000000 +0100
+++ ../atari800/src/avic.c	2013-03-04 14:55:48.673849800 +0100
@@ -0,0 +1,89 @@
+#include "config.h"
+#include <stdio.h>
+#include <string.h>
+#include "avic.h"
+#include "log.h"
+#include "memory.h"
+
+static UBYTE bootstrap[256];
+static FILE *sdcard = NULL;
+
+static FILE *AVIC_Open(const char *filename)
+{
+	FILE *fp = fopen(filename, "rb");
+	if (fp == NULL)
+		Log_print("Cannot open %s", filename);
+	return fp;
+}
+
+static int AVIC_Read(FILE *fp, UBYTE *buffer, size_t len)
+{
+	if (fread(buffer, len, 1, fp) != 1) {
+		Log_print("AVIC file too short");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+int AVIC_Initialise(int *argc, char **argv)
+{
+	int i;
+	int j;
+	for (i = j = 1; i < *argc; i++) {
+		char *arg = argv[i];
+		if (strcmp(arg, "-avic") == 0) {
+			FILE *fp = AVIC_Open(argv[++i]);
+			int ok;
+			if (fp == NULL)
+				return FALSE;
+			ok = AVIC_Read(fp, bootstrap, 0x100);
+			fclose(fp);
+			if (!ok)
+				return FALSE;
+			sdcard = AVIC_Open(argv[++i]);
+			if (sdcard == NULL)
+				return FALSE;
+		}
+		else {
+			if (strcmp(arg, "-help") == 0)
+				Log_print("\t-avic BOOTSTRAP.RAW SDCARD.RAW  Emulate Atari VIdeo Cartridge");
+			argv[j++] = arg;
+		}
+	}
+	*argc = j;
+	return TRUE;
+}
+
+void AVIC_Coldstart(void)
+{
+	if (sdcard == NULL)
+		return;
+	MEMORY_SetHARDWARE(0x9ffc, 0x9ffc);
+	MEMORY_CartA0bfEnable();
+	MEMORY_dCopyToMem(bootstrap, 0xbf00, 0x100);
+}
+
+int AVIC_PutByte(UBYTE byte)
+{
+	UWORD address;
+	long offset;
+	if (sdcard == NULL)
+		return FALSE;
+	switch (byte & 0x11) {
+	case 0x01:
+		address = 0x8000;
+		break;
+	case 0x10:
+		address = 0xa000;
+		break;
+	default:
+		return TRUE;
+	}
+	offset = (MEMORY_dGetByte(0x9ffd) << 8 | MEMORY_dGetByte(0x9ffe) << 16 | MEMORY_dGetByte(0x9fff) << 24) - 0x3cd600;
+	/* TODO: read isn't immediate */
+	if (fseek(sdcard, offset, SEEK_SET) != 0)
+		Log_print("AVIC seek error");
+	else
+		AVIC_Read(sdcard, MEMORY_mem + address, 0x1ffc);
+	return TRUE;
+}
diff -ruN -x Makefile -x '*.o' -x 'atari800*' -x autom4te.cache -x 'config*' -x atari.ro ../../emusrc/atari800/src/avic.h ../atari800/src/avic.h
--- ../../emusrc/atari800/src/avic.h	1970-01-01 01:00:00.000000000 +0100
+++ ../atari800/src/avic.h	2013-03-04 14:24:30.738438100 +0100
@@ -0,0 +1,10 @@
+#ifndef AVIC_H_
+#define AVIC_H_
+
+#include "atari.h"
+
+int AVIC_Initialise(int *argc, char **argv);
+void AVIC_Coldstart(void);
+int AVIC_PutByte(UBYTE byte);
+
+#endif
diff -ruN -x Makefile -x '*.o' -x 'atari800*' -x autom4te.cache -x 'config*' -x atari.ro ../../emusrc/atari800/src/memory.c ../atari800/src/memory.c
--- ../../emusrc/atari800/src/memory.c	2012-11-05 15:06:29.260712700 +0100
+++ ../atari800/src/memory.c	2013-03-04 14:23:47.764980100 +0100
@@ -40,6 +40,7 @@
 #include "pia.h"
 #include "pokey.h"
 #include "util.h"
+#include "avic.h"
 #ifndef BASIC
 #include "statesav.h"
 #endif
@@ -1151,6 +1152,8 @@
 		break;
 	case 0x5f00:
 	case 0x9f00:
+		if (AVIC_PutByte(byte))
+			break;
 		CARTRIDGE_BountyBob2(addr);
 		break;
 	case 0xd000:				/* GTIA */
